cpp/src/gandiva/decimal_scalar.h

```diff
@@ -55,3 +55,19 @@ class DecimalScalar128 : public BasicDecimalScalar128 {
 };
 
 }  // namespace gandiva
+
+namespace std {
+template <>
+struct hash<gandiva::DecimalScalar128> {
+  std::size_t operator()(gandiva::DecimalScalar128 const& s) const noexcept {
+    arrow::BasicDecimal128 dvalue(s.value());
+    std::size_t h0 = std::hash<int64_t>{}(dvalue.high_bits());
+    std::size_t h1 = std::hash<uint64_t>{}(dvalue.low_bits());
+
+    std::size_t h2 = std::hash<int32_t>{}(s.precision());
+    std::size_t h3 = std::hash<int32_t>{}(s.scale());
+
+    return (((h0 ^ (h1 << 1) >> 1) ^ (h2 << 1) >> 1) ^ h3 << 1);
```

Comment:
why not use arrow:util:hash_combine

Reply:
I didn't know about it yet, can you just check if I should hash the values before I pass then to the hash_combine function?

Reply:
No. Take a look at https://github.com/apache/arrow/blob/master/cpp/src/gandiva/filter.cc#L48

Reply:
Ok, I already had saw it, gonna change it right now